src/pages/api/auth/callback.ts

```diff
@@ -0,0 +1,42 @@
+import { NextApiRequest, NextApiResponse } from 'next'
+
+import { getAuthorizationCode, getClientCredentials } from '@/auth/oauth2'
+
+export default async function callback(
+  req: NextApiRequest,
+  res: NextApiResponse
+) {
+  const oauth2ClientCredentials = getClientCredentials()
+  const oauth2AuthorizationCode = getAuthorizationCode()
+
+  if (oauth2ClientCredentials === null || oauth2AuthorizationCode === null) {
+    res.status(500)
+    res.send('Auth not configured correctly')
+    return
+  }
+
+  const { code, scope, state } = req.query as {
+    code: string
+    scope: string
+    state: string
+  }
+  const { csrf, referer } = JSON.parse(state)
+  if (csrf === req.cookies['auth-csrf']) {
+    const result = await oauth2AuthorizationCode.getToken({
+      code,
+      redirect_uri: 'http://localhost:3000/api/auth/callback',
+      scope,
+    })
+    const { token } = oauth2ClientCredentials.createToken(result)
```

Comment:
Just wondering: why are we not using `oauth2AuthorizationCode.createToken()` here?

Reply:
Intended, the types are not up-to-date and this should be `clientCredentials` stuff